src/main/java/com/jagrosh/giveawaybot/Bot.java

```diff
@@ -108,16 +110,26 @@ public boolean startGiveaway(TextChannel channel, User creator, Instant now, int
             return false;
         database.settings.updateColor(channel.getGuild());
         Instant end = now.plusSeconds(seconds);
-        Message msg = new Giveaway(0, channel.getIdLong(), channel.getGuild().getIdLong(), creator.getIdLong(), end, winners, prize, Status.RUN, false)
+        String emoji = database.settings.getSettings(channel.getGuild().getIdLong()).getEmojiRaw();
+        Message msg = new Giveaway(0, channel.getIdLong(), channel.getGuild().getIdLong(), creator.getIdLong(), end, winners, prize, emoji, Status.RUN, false)
                 .render(channel.getGuild().getSelfMember().getColor(), now);
-        channel.sendMessage(msg).queue(m -> 
-        {
-            m.addReaction(Constants.TADA).queue();
-            database.giveaways.createGiveaway(m, creator, end, winners, prize, false);
+
+        if (emoji == null || emoji.isEmpty())
+            emoji = Constants.TADA;
+
+        final AtomicReference<String> finalEmoji = new AtomicReference<>(emoji);
+        channel.sendMessage(msg).queue(m -> {
+            m.addReaction(finalEmoji.get()).onErrorFlatMap(ignored -> { // this might be perms error or because we can't add that emoji
+                channel.sendMessageFormat("%s Failed to use your custom emoji. It's been automatically reset.\nThis message self destructs in 20 seconds.", Constants.WARNING).delay(20, TimeUnit.SECONDS).flatMap(Message::delete).queue(s->{},f->{});
+                database.settings.updateEmoji(channel.getGuild(), null);
+                finalEmoji.set(null);
+                return m.addReaction(Constants.TADA);
+            }).queue();
+            database.giveaways.createGiveaway(m, creator, end, winners, prize, finalEmoji.get(), false);
```

Comment:
I'm not sure this section will actually do what you intend it to; `finalEmoji.set(null)` may run after the giveaway has already been created in the database, causing the giveaway to run incorrectly.

Reply:
Because of this and other string-related issues, I think we may want to add an `Emoji` class to the settings, something along the lines of:
```java
class Emoji {
    private final String raw;
    public Emoji(String value) {
        this.raw = value;
    }
    public String getReaction() {
        return raw == null ? TADA : raw;
    }
    public String getDisplay() {
        return raw == null ? TADA : raw.contains(":") ? "<" + raw + ">" : raw;
    }
}
````
This way we can hold and work with null values without having to check all the time, and it makes it easier for insertion into the database as well.